crypto/rsa/rsa_pmeth.c

```diff
@@ -864,7 +883,12 @@ static int pkey_pss_init(EVP_PKEY_CTX *ctx)
         return 0;
 
     /* See if minimum salt length exceeds maximum possible */
-    max_saltlen = RSA_size(rsa) - EVP_MD_get_size(md);
+    md_size = EVP_MD_get_size(md);
+    if (md_size <= 0) {
+        ERR_raise(ERR_LIB_RSA, RSA_R_INVALID_DIGEST_LENGTH);
+        return 0;
+    }
+    max_saltlen = RSA_size(rsa) - md_size;
```

Comment:
Should we check for `RSA_size(rsa) < md_size` to avoid a negative `max_saltlen`?

Reply:
I find there is already a check afterward "if (min_saltlen > max_saltlen)." And only if "max_saltlen" is MIN_INT can an overflow occur in the "if ((RSA_bits(rsa) & 0x7) == 1) max_saltlen--;". However, subtracting two positive numbers cannot result in MIN_INT.